ec/ecresource-tpf/deploymentresource/deployment.go

```diff
@@ -0,0 +1,146 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentresource
+
+import (
+	"fmt"
+
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/elastic/terraform-provider-ec/ec/internal/flatteners"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type Deployment struct {
+	Id                    types.String          `tfsdk:"id"`
+	Alias                 types.String          `tfsdk:"alias"`
+	Version               types.String          `tfsdk:"version"`
+	Region                types.String          `tfsdk:"region"`
+	DeploymentTemplateId  types.String          `tfsdk:"deployment_template_id"`
+	Name                  types.String          `tfsdk:"name"`
+	RequestId             types.String          `tfsdk:"request_id"`
+	ElasticsearchUsername types.String          `tfsdk:"elasticsearch_username"`
+	ElasticsearchPassword types.String          `tfsdk:"elasticsearch_password"`
+	ApmSecretToken        types.String          `tfsdk:"apm_secret_token"`
+	TrafficFilter         []string              `tfsdk:"traffic_filter"`
+	Tags                  types.Map             `tfsdk:"tags"`
+	Elasticsearch         []*Elasticsearch      `tfsdk:"elasticsearch"`
+	Kibana                []*Kibana             `tfsdk:"kibana"`
+	Apm                   []*Apm                `tfsdk:"apm"`
+	IntegrationsServer    []*IntegrationsServer `tfsdk:"integrations_server"`
+	EnterpriseSearch      []*EnterpriseSearch   `tfsdk:"enterprise_search"`
+	Observability         []*Observability      `tfsdk:"observability"`
```

Comment:
>The recommended way to deal with configuration or plan data for now is using types package types in Go structs, e.g.
> ```
> type myModel struct {
>   ExampleList types.List `tfsdk:"example_list"`
> }
> ```
-- https://discuss.hashicorp.com/t/terraform-plugin-framework-value-conversion-error-unhandled-unknown-value/44382/3

Reply:
Yeah, it's a valid concern. It looks like there is also an option to leverage [appropriate interfaces](https://www.terraform.io/plugin/framework/accessing-values#detected-interfaces). 
I'll reconsider these definitions when I complete `Create`/`Update`.
Thank you for the heads-up.